src/ExceptionTypes.f90

```diff
@@ -0,0 +1,370 @@
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!                          Futility Development Group                          !
+!                             All rights reserved.                             !
+!                                                                              !
+! Futility is a jointly-maintained, open-source project between the University !
+! of Michigan and Oak Ridge National Laboratory.  The copyright and license    !
+! can be found in LICENSE.txt in the head directory of this repository.        !
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!> @brief Utility module for defining exception types used by the
+!>  ExceptionHandler
+!>
+!> @author William Gurecky
+!>   @date 12/16/2019
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+MODULE ExceptionTypes
+#include "Futility_DBC.h"
+  USE Futility_DBC
+  USE ISO_FORTRAN_ENV
+  USE IntrType
+
+  IMPLICIT NONE
+  PRIVATE !Default private for module contents
+!
+! List of Public items
+  PUBLIC :: ExceptionTypeBase
+  PUBLIC :: ExceptionTypeError
+  PUBLIC :: ExceptionTypeWarning
+  PUBLIC :: ExceptionTypeFatal
+  PUBLIC :: ExceptionTypeDebug
+  PUBLIC :: ExceptionTypeInformation
+  PUBLIC :: EXCEPTION_MAX_MESG_LEN
+
+  !> The maximum size of an exception message
+  INTEGER(SIK),PARAMETER :: EXCEPTION_MAX_MESG_LEN=512
+
+  TYPE, ABSTRACT :: ExceptionTypeBase
```

Comment:
Another thing we've moved away from stylisticly is having the word `Type` in extended types.  It's pretty well never helpful and just makes things wordier.  Do some search and replace to change `ExceptionType` to `Exception`, basically everywhere.